docs/dump_psi_to_file.md

```diff
@@ -0,0 +1,9 @@
+## Dumping the PSI to a file
```

Comment:
It's probably worth noting that this is about R# trees, not about IntelliJ ones. Maybe we should have separate docs (separate folders?) for these two plugins, and an additional doc telling which plugin does what.

It's not relevant just yet, but once https://github.com/JetBrains/resharper-fsharp/pull/482 is merged, it'd be helpful to have docs about dumping those trees as well. It can be done like this (and it requires [Internal mode](https://rider-support.jetbrains.com/hc/en-us/articles/207327910-How-to-run-JetBrains-Rider-in-Internal-Mode)):

<img width="549" alt="Screenshot 2023-03-02 at 11 32 41" src="https://user-images.githubusercontent.com/3923587/222403611-c29bafb3-1deb-4aa2-a281-8e2d28edc860.png">

Reply:
> It's probably worth noting that this is about R# trees, not about IntelliJ ones. Maybe we should have separate docs (separate folders?) for these two plugins, and an additional doc telling which plugin does what.
> 
Okay, I moved the new `docs` folder down to `ReSharper.FSharp`.
About the distinction of the two plugins, I think that's already covered in the existing `README.md`.

> It's not relevant just yet, but once #482 is merged, it'd be helpful to have docs about dumping IntelliJ trees as well. It can be done like this (and it requires [Internal mode](https://rider-support.jetbrains.com/hc/en-us/articles/207327910-How-to-run-JetBrains-Rider-in-Internal-Mode), it's automatically used when running the plugin via Gradle):

Sure, once that's done, I can provide docs for that, too :)